File: server/Bank.Core/JsonModels/Bank/CreditConditions/CreditTypeConditionModel.cs
using Bank.Core.Enums;

namespace Bank.Core.JsonModels.Bank.CreditConditions;

public class CreditTypeConditionModel
{
    public long Id { get; set; }
    public CreditType CreditType { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal StandardAnnualInterestRate { get; set; }
    public decimal VipAnnualInterestRate { get; set; }
    public decimal MaximumAmount { get; set; }
    public int MaximumTermMonths { get; set; }
    public decimal StandardGrantingFee { get; set; }
    public decimal VipGrantingFee { get; set; }
    public PaymentType DefaultPaymentType { get; set; }
    public int PromoPeriodMonths { get; set; }
    public decimal? StandardPromoRate { get; set; }
    public decimal? VipPromoRate { get; set; }
    public int GracePeriodMonths { get; set; }
    public decimal StandardMonthlyManagementFee { get; set; }
    public decimal VipMonthlyManagementFee { get; set; }
    public decimal StandardAnnualManagementFee { get; set; }
    public decimal VipAnnualManagementFee { get; set; }
    public bool IsActive { get; set; }
}

File: client/src/types/JsonModels/Bank/CreditConditions/CreditTypeConditionModel.ts
import type { CreditType } from "../../Enums/CreditType";
import type { PaymentType } from "../../Enums/PaymentType";

export interface CreditTypeConditionModel
{
	id: number;
	creditType: CreditType;
	name: string;
	standardAnnualInterestRate: number;
	vipAnnualInterestRate: number;
	maximumAmount: number;
	maximumTermMonths: number;
	standardGrantingFee: number;
	vipGrantingFee: number;
	defaultPaymentType: PaymentType;
	promoPeriodMonths: number;
	standardPromoRate?: number;
	vipPromoRate?: number;
	gracePeriodMonths: number;
	standardMonthlyManagementFee: number;
	vipMonthlyManagementFee: number;
	standardAnnualManagementFee: number;
	vipAnnualManagementFee: number;
	isActive: boolean;
}
